packages/model-viewer/src/three-components/SmoothControls.ts

```diff
@@ -839,33 +878,121 @@ export class SmoothControls extends EventDispatcher {
+    this.panScale = (event.ctrlKey || event.key == 'Control') ? this.panMouseMultiplier : PAN_SCALE
+
+    if (relevantKey) {
+      event.preventDefault();
+    }
+  };
+
+  /**
+   * Hide $panElement when the shift key or (when shift is a modifier) is the key that has "upped"
+   * Only testing shift key misses then case when shift is a modifier
+   * Always setting the opacity to 0 means $panElement winks on and off
+   * Reset panScale on every KeyUp event
+   * @param event The keyboard event for the .key and .shiftKey value
+   */
+  private onKeyUp = (event: KeyboardEvent) => {
+    (this.scene.element as any)[$panElement].style.opacity = (event.key == 'Shift' || !event.shiftKey) ? 0:1;
+    this.panScale = PAN_SCALE;
+  }
+
+  /**
+   * Handles the orbit and Zoom key presses
+   * @param event The keyboard event for the .key value
+   * @param ZoomInc The zoom increment
+   * @param OrbitInc The orbit increment
+   * @param isUserChange Is this a user initiated event
+   * @returns boolean to indicate if the key event has been handled
+   */
+  private orbitZoomKeyCodeHandler(event: KeyboardEvent, ZoomInc: number, OrbitInc: number, isUserChange: boolean) {
+    // releventKey as the return value is to get the condition back to the calling function as don't have pass by reference arguments in js
+    let relevantKey = true;
+    switch (event.key) {
+      case 'PageUp':
+        this.userAdjustOrbit(0, 0, ZoomInc);
         break;
-      case KeyCode.PAGE_DOWN:
-        this.userAdjustOrbit(0, 0, -1 * ZOOM_SENSITIVITY);
+      case 'PageDown':
+        this.userAdjustOrbit(0, 0, -1 * ZoomInc);
         break;
-      case KeyCode.UP:
-        this.userAdjustOrbit(0, -KEYBOARD_ORBIT_INCREMENT, 0);
+      case 'ArrowUp':
+        this.userAdjustOrbit(0, -OrbitInc, 0);
         break;
-      case KeyCode.DOWN:
-        this.userAdjustOrbit(0, KEYBOARD_ORBIT_INCREMENT, 0);
+      case 'ArrowDown':
+        this.userAdjustOrbit(0, OrbitInc, 0);
         break;
-      case KeyCode.LEFT:
-        this.userAdjustOrbit(-KEYBOARD_ORBIT_INCREMENT, 0, 0);
+      case 'ArrowLeft':
+        this.userAdjustOrbit(-OrbitInc, 0, 0);
         break;
-      case KeyCode.RIGHT:
-        this.userAdjustOrbit(KEYBOARD_ORBIT_INCREMENT, 0, 0);
+      case 'ArrowRight':
+        this.userAdjustOrbit(OrbitInc, 0, 0);
         break;
       default:
         relevantKey = false;
         this.isUserChange = isUserChange;
         break;
     }
 
-    if (relevantKey) {
-      event.preventDefault();
+    return relevantKey;
+  }
+
+ /**
+  * Handles the Pan key presses
+  * @param event The keyboard event for the .key value
+  * @param dxy The magnitude of the move
+  * @param isUserChange Is this a user initiated event
+  * @returns boolean to indicate if the key event has been handled
+  */
+    private panKeyCodeHandler(event: KeyboardEvent, dxy: number, isUserChange: boolean) {
```

Comment:
The problem with reviewing this PR is it's too big and doing too many things. Let's split it up and work up slowly. Start with a PR that just adds keyboard pan functionality. As small a change as possible; no API change, no multiple rates. 

Then, you can make a separate one to make the keyboard sensitivity based on our existing sensitivity parameter. There we can experiment with the out-of-sync issues you were talking about and see if we can fix them. After that, we can look into the rest in separate PRs.

Does that sound good? Also a new PR is probably the best way to get rid of the weird diff stuff in this one.

Reply:
Yes ok. 
Now to confirm, before clicking on the wrong thing yet again, I should close this pull request and start a new one?

(I've worked out that you can only have one PR open at anyone time)

Reply:
Yes, let's go ahead and close this PR and open a new one (or possibly several). You are definitely allowed to open as many PRs as you like simultaneously. In practice, that's only a good idea if they are independent (no major merge conflicts between them).